src/main/java/com/google/sps/servlets/DashboardServlet.java

```diff
@@ -14,62 +14,125 @@
 import com.google.appengine.api.datastore.Query.Filter;
 import com.google.appengine.api.datastore.Query.FilterPredicate;
 import com.google.appengine.api.datastore.Query.FilterOperator;
+import com.google.appengine.api.datastore.Query.CompositeFilterOperator;
 import com.google.sps.servlets.Property;
 import java.lang.reflect.Field; 
 import java.util.HashMap;
+import com.google.sps.servlets.RequestUtils;
 
 @WebServlet("/dashboard")
 public class DashboardServlet extends HttpServlet {
 
+  private final long MILLISECONDS_IN_DAY = 86400000;
+  private final String DEFAULT_START_TIMESTAMP = "0";
+  private final String DEFAULT_END_TIMESTAMP = String.valueOf(System.currentTimeMillis());
+  private String startTimestamp;
+  private String endTimestamp;
+
  /**
   * This route will return a JSON with the percentages of each interaction
   */
   @Override
   public void doGet(HttpServletRequest request, HttpServletResponse response) throws IOException {
 
+    String requestStartTimestamp = RequestUtils.getParameter(request, "startTime");
+    String requestEndTimestamp = RequestUtils.getParameter(request, "endTime");
+
+    validateAndSetVariables(requestStartTimestamp, requestEndTimestamp);    
+    
     try {
-      HashMap<String, Double> interactionPercentages = calculatePercentages();
+      HashMap<String, Double> dataToSend = calculatePercentages();
       
-      String jsonToSend = convertToJson(interactionPercentages);
+      String jsonToSend = convertToJson(dataToSend);
 
+      response.addHeader("Access-Control-Allow-Origin", "*");
```

Comment:
Did you confirm that this is actually needed? Ideally, if we can not include this, or at least set it to just the specific domain that needs it, it would be alot better than opening it up to anyone

Reply:
Removed it and it still works - i think